monitor/monitor.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"errors"
+	"os"
+
+	"github.com/cloudwego/kitex/pkg/klog"
+	"github.com/kitex-contrib/config-file/parser"
+	"github.com/kitex-contrib/config-file/utils"
+)
+
+type Options struct {
+	FilePath string      // config file path
+	Key      string      // config key
+	Provider KeyProvider // custom key provider
+}
+
+type ConfigMonitor struct {
+	manager     parser.ConfigManager // Manager for the config file
+	config      interface{}          // config details
+	fileWatcher *utils.FileWatcher   // local config file watcher
+	callbacks   []func()             // callbacks when config file changed
+	key         string               // key
+}
+
+// NewConfigMonitor init a monitor for the config file
+func NewConfigMonitor(opts Options) (*ConfigMonitor, error) {
+	if opts.FilePath == "" {
+		return nil, errors.New("empty config file path")
+	}
+	if opts.Key == "" {
+		return nil, errors.New("empty config key")
+	}
+	if opts.Provider == nil {
+		opts.Provider = NewDefaultKeyProvider()
+	}
+
+	fw, err := utils.NewFileWatcher(opts.FilePath)
```

Comment:
So we're still creating one FileWatcher for each client? If we have 100 clients and once the config file is changed, it will be read and parsed 100 times.